sentry_sdk/profiler.py

```diff
@@ -791,12 +795,7 @@ def make_sampler(self):
         # type: () -> Callable[..., None]
         cwd = os.getcwd()
 
-        # In Python3+, we can use the `nonlocal` keyword to rebind the value,
-        # but this is not possible in Python2. To get around this, we wrap
-        # the value in a list to allow updating this value each sample.
-        last_sample = [
-            {}
-        ]  # type: List[Dict[int, Tuple[RawStackId, RawStack, Deque[FrameType]]]]
+        cache = LRUCache(max_size=256)
```

Comment:
Need to assess what the right size for this cache is.

Reply:
256 should be a sufficiently large cache, we want a cache larger than the stack limit of 128 so if 2 unique stacks are captured, it doesn't completely invalidate the cache. At 256, it should have enough room for the common frames to be kept in the cache.